validator/coordinator/src/main/java/tech/pegasys/artemis/validator/coordinator/Eth1DataCache.java

```diff
@@ -113,13 +80,34 @@ public Eth1Data get_eth1_vote(BeaconState state) {
     return vote.orElse(defaultVote);
   }
 
-  private NavigableMap<UnsignedLong, Eth1Data> getVotesToConsider() {
-    return eth1ChainCache.subMap(getSpecRangeLowerBound(), true, getSpecRangeUpperBound(), true);
+  private NavigableMap<UnsignedLong, Eth1Data> getVotesToConsider(final UnsignedLong slot) {
+    return eth1ChainCache.subMap(
+        getSpecRangeLowerBound(slot), true, getSpecRangeUpperBound(slot), true);
+  }
+
+  private UnsignedLong calculateCacheDuration() {
+    // Worst case we're in the very last moment of the current slot
+    long cacheDurationSeconds = SECONDS_PER_SLOT;
+
+    // Worst case this slot is at the very end of the current voting period
+    cacheDurationSeconds += (EPOCHS_PER_ETH1_VOTING_PERIOD * SLOTS_PER_EPOCH * SECONDS_PER_SLOT);
+
+    // We need 2 * ETH1_FOLLOW_DISTANCE prior to that
+    cacheDurationSeconds +=
+        SECONDS_PER_ETH1_BLOCK.longValue() * ETH1_FOLLOW_DISTANCE.longValue() * 2;
+
+    // And we want to be able to create blocks for at least the past epoch
```

Comment:
This one I do not understand the reasoning well but I do get that a buffer is necessary anyway.

Reply:
The validator APIs allow the validator to specify the slot they need to create a block for, so there's no guarantee that the block we're building is at our current node slot - it could be somewhere before that (or after but that doesn't affect this).